### DISCUSSION TITLE
Python_ Can support VN1670?

### DISCUSSION THREAD

**Discussion by waynechenjun** [hardbyte/python-can#1619] [Q&A]
device_name : VN1670

bus = can.interface.Bus(bustype='vector', app_name="CANoe", channel=0,bitrate=500000)

Traceback (most recent call last):
  
    bus = can.interface.Bus(bustype='vector', app_name="CANoe", channel=0,bitrate=500000)

  File "C:\Python\Python38\lib\site-packages\can\interface.py", line 127, in __new__
    return cls(channel, *args, **kwargs)
  File "C:\Python\Python38\lib\site-packages\can\interfaces\vector\canlib.py", line 134, in __init__
    raise VectorError(vxlapi.XL_ERR_HW_NOT_PRESENT,

can.interfaces.vector.exceptions.VectorError: xlGetChannelIndex failed (XL_ERR_HW_NOT_PRESENT)

How to modify？？

**Comment by JacobFEV** [comment]
My advice would be to try to set up your bus without passing an "app_name" into it. I have been able to connect to the VN1610 and the VN1630A by doing this:

```
import can
bus = can.Bus(bustype="vector", channel=0, bitrate=500000)
```

You could also try reading the serial number of the device and pass that into the bus creation as such:

```
import can
bus = can.Bus(bustype="vector", channel=0, bitrate=500000, serial=<insert serial number here>)
```

The second example is most useful for connecting to multiple buses at the same time. See if these help to connect the VN1670.
